homeassistant/components/agent_dvr/camera.py

```diff
@@ -0,0 +1,263 @@
+
+        device_info = {
+            CONF_NAME: device.name,
+            CONF_MJPEG_URL: f"{self.server_url}{device.mjpeg_image_url}&size=640x480",
+            CONF_STILL_IMAGE_URL: f"{self.server_url}{device.still_image_url}&size=640x480",
+        }
+        self.device = device
+        self._removed = False
+        self._name = f"{self._servername} {device.name}"
+        self._stream_url = f"{self.server_url}{device.mp4_url}"
+        self._unique_id = f"{device._client.unique}_{device.typeID}_{device.id}"
+        self._enabled_default = enabled_default
+        super().__init__(device_info)
+
+    @property
+    def device_info(self):
+        """Return the device info for adding the entity to the agent object."""
+        return {
+            "identifiers": {(AGENT_DOMAIN, self._unique_id)},
+            "name": self._name,
+            "manufacturer": "Agent",
+            "model": "Camera",
+            "sw_version": self.device.client.version,
+        }
+
+    async def async_update(self):
+        """Update our state from the Agent API."""
+        try:
+            await self.device.update()
+            if self._removed:
+                _LOGGER.error("%s reacquired", self._name)
+            self._removed = False
+        except AgentError:
+            if self.device.client.is_available:  # server still available - camera error
+                if not self._removed:
+                    _LOGGER.error("%s lost", self._name)
+                    self._removed = True
+
+    @property
+    def state_attributes(self):
+        """Return the Agent DVR camera state attributes."""
+        return {
+            ATTR_ATTRIBUTION: ATTRIBUTION,
+            "editable": False,
+            "enabled": self.enabled,
+            "connected": self.connected,
+            "detected": self.is_detected,
+            "alerted": self.is_alerted,
+            "recording": self.is_recording,
+            "has_ptz": self.device.has_ptz,
+            "motion_detection_enabled": self.device.detector_active,
+            "alerts_enabled": self.device.alerts_active,
+        }
+
+    @property
+    def is_recording(self) -> bool:
+        """Return whether the monitor is recording."""
+        return self.device.recording
+
+    @property
+    def is_alerted(self) -> bool:
+        """Return whether the monitor has alerted."""
+        return self.device.alerted
+
+    @property
+    def is_detected(self) -> bool:
+        """Return whether the monitor has alerted."""
+        return self.device.detected
+
+    @property
+    def available(self) -> bool:
+        """Return True if entity is available."""
+        return self.device.client.is_available
+
+    @property
+    def connected(self) -> bool:
+        """Return True if entity is connected."""
+        return self.device.connected
+
+    @property
+    def enabled(self) -> bool:
+        """Return True if entity is enabled."""
+        return self.device.online
```

Comment:
We return the same thing in `is_on` property. We can remove the `enabled` property.